From 02e57c2192d2772ac5da7558d98510426c5fc14e Mon Sep 17 00:00:00 2001
From: Mrinmoy Ghosh <mrghosh@cisco.com>
Date: Tue, 19 Aug 2025 22:52:23 +0000
Subject: [PATCH 2/9] zebra: New protocol RTPROT_HW for hardware learnt MAC

Protocol field is added in bridge FDB, to distinguish between
 MAC addresses learned via the control plane and those learned
 via the data plane with hardware aging.

Protocol 'hw' (i.e RTPROT_HW aka hardware) for MAC learnt by hardware
 will be used for data plane(hardware) learnt MAC while
 existing protocol 'zebra' to be used for control plane learnt ones.

Kernel Patch in review:
https://lore.kernel.org/netdev/20250818175258.275997-1-mrghosh@cisco.com/
iproute2 patch in review:
https://lore.kernel.org/netdev/20250818193756.277327-1-mrghosh@cisco.com/

Signed-off-by: Mrinmoy Ghosh <mrinmoy_g@hotmail.com>
Signed-off-by: Mrinmoy Ghosh <mrghosh@cisco.com>
Signed-off-by: Patrice Brissette <pbrisset@cisco.com>
Signed-off-by: Tamer Ahmed <tamerahmed@microsoft.com>
---
 include/linux/rtnetlink.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/rtnetlink.h b/include/linux/rtnetlink.h
index d03ed4d16e..5421794141 100644
--- a/include/linux/rtnetlink.h
+++ b/include/linux/rtnetlink.h
@@ -308,7 +308,7 @@ enum {
 #define RTPROT_OSPF		188	/* OSPF Routes */
 #define RTPROT_RIP		189	/* RIP Routes */
 #define RTPROT_EIGRP		192	/* EIGRP Routes */
-
+#define RTPROT_HW		193	/* Hardware Routes */
 /* rtm_scope
 
    Really it is not scope, but sort of distance to the destination.
-- 
2.43.0

